operator/v1/types_klusterlet.go

```diff
@@ -86,6 +86,10 @@ type KlusterletSpec struct {
 	// registration-agent and work-agent will use it to communicate with hub api server.
 	// +optional
 	HubApiServerHostAlias *HubApiServerHostAlias `json:"hubApiServerHostAlias,omitempty"`
+
+	// BestEffortResource allow use the best effort resource for operator/registration/work deployment
+	// +optional
+	BestEffortResource bool `json:"bestEffortResource,omitempty"`
```

Comment:
It is better no to use boolean in API. You might consider a resources struct with a type in it. We can just set type to "BestEffort" and "Default" today, and add other option to be able to specify resource in the future.

Reply:
@qiujian16  thanks. address your comments and PTAL again.